pkg/util/configmap_util.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// Ordering of elements in configmap key
+// key is in form {projectID}.{location}.{filestoreName}.{shareName}.{gkeNodeID}.{gkeNodeInternalIP}
+// Adding a new element should always go at the end
+const (
+	projectID = iota
+	location
+	filestoreName
+	shareName
+	gkeNodeID         // GCE instance ID for the GKE node
+	gkeNodeInternalIP // GKE node internal IP concatenated by underscores
+	totalKeyElements  // Always last
+)
+
+const (
+	ConfigMapNamePrefix = "fscsi-"
+	ConfigMapNamespace  = "gke-managed-filestorecsi"
+	ConfigMapFinalzer   = "filestore.csi.storage.gke.io/lock-release"
```

Comment:
Can you put a comment on how the finalizer is intended to be used

Reply:
Done